### PR TITLE
Adding new post on token auth in Node. Also adding Macy to authors file.

### PR DESCRIPTION
**Pull request by rdegges**
This is a new post that should go live on Valentine's Day (2/14). Loading it up early. @mraible would you give this a quick review for me? Thanks! <3

### CODE REVIEW

File: _source/_posts/2019-02-14-modern-token-authentication-in-node-with-express.md [hunk 1]
```diff
@@ -0,0 +1,356 @@
+---
+layout: blog_post
+title: "Modern Token Authentication in Node with Express"
+author: mngan
+description: "Come learn all about what token authentication is and how to use it in your Node + Express apps."
+tags: [jwt, node, javascript, express, token, authentication]
+tweets:
+  - "Check this out--we just published a new article detailing how token auth
```
Comment (mraible) [inline _source/_posts/2019-02-14-modern-token-authentication-in-node-with-express.md]: Does it matter if this tweet isn't all on one line?

File: _source/_posts/2019-02-14-modern-token-authentication-in-node-with-express.md [hunk 2]
```diff
@@ -0,0 +1,356 @@
...
+
+JWTs are the most popular type of tokens and are often what people mean when they refer to *"token authentication"* in general.
+
+Here's what a typical JWT might look like in it's compacted, URL-safe form:
+`eyJraWQiOiJ1dURLVTMxZWRvTi0wd0xMUnl1TW1vbmtBdi1OaFEwejZhWmxjdTN5NU8wIiwiYWxnIjoiUlMyNTYifQ.eyJ2ZXIiOjEsImp0aSI6IkFULnVfT094R3pXd1RjRFlseGZwcDVYXzNxdVIwdlJuc25YbXdMZld0TDFjdG8iLCJpc3MiOiJodHRwczovL2Rldi04MTk2MzMub2t0YXByZXZpZXcuY29tL29hdXRoMi9kZWZhdWx0IiwiYXVkIjoiYXBpOi8vZGVmYXVsdCIsImlhdCI6MTU0NjcyNjIyOCwiZXhwIjoxNTQ2NzI5ODI4LCJjaWQiOiIwb2Fpb3g4Ym1zQktWWGt1MzBoNyIsInNjcCI6WyJjdXN0b21TY29wZSJdLCJzdWIiOiIwb2Fpb3g4Ym1zQktWWGt1MzBoNyJ9.fhZRWThFxhiS6Pgrup6hM08oSqDPd1JrZSDIH_blD5S20c2hQQ3D3RZyhNKMnYclyus_mo-H-mS-Ak3YzM8S0JwZ8m3Vid4smW953peBMnmBNotAE--yE0toc2dIUG3BWQR34hah253bKmp17Yh6bWGwH60oQxnuM_NVUpa-NJMBc6-Mu5ra0lKfr60ne9-jKVFcavd9ZnVTLiug_sXFlhxgaQm4V_hhcvcLSwCXTiIcQsJkI0rP7WuVvjYVyK_sPeW3A44_T5qhyDR_E_mk1rHORlkMYGPg34mcwob5iA7alNZOnzN_7ApcbylDbK5KS1umBqqevtghEyjOEWQQmQ`
```
Comment (mraible) [inline _source/_posts/2019-02-14-modern-token-authentication-in-node-with-express.md]: Would this look better if it was in a code block? I'm not sure if this long string will have line breaks in it or not.

File: _source/_posts/2019-02-14-modern-token-authentication-in-node-with-express.md [hunk 3]
```diff
@@ -0,0 +1,356 @@
...
+cd authapp
+```
+
+Initialize the project:
+
+```bash
+npm init
+```
+
+Keep hitting enter to accept all default settings.
```
Comment (mraible) [inline _source/_posts/2019-02-14-modern-token-authentication-in-node-with-express.md]:
```suggestion
Keep hitting enter to accept all the default settings.
```

File: _source/_posts/2019-02-14-modern-token-authentication-in-node-with-express.md [hunk 4]
```diff
@@ -0,0 +1,356 @@
...
+
+Install Express:
+
+```bash
+npm install express@4.16.4 --save
```
Comment (mraible) [inline _source/_posts/2019-02-14-modern-token-authentication-in-node-with-express.md]:
```suggestion
npm install express@4.16.4
```
`--save` is no longer needed in npm5 + https://twitter.com/maybekatz/status/859229741676625920

File: _source/_posts/2019-02-14-modern-token-authentication-in-node-with-express.md [hunk 5]
```diff
@@ -0,0 +1,356 @@
...
+```
+
+Install the [Okta JWT Verifier for Node.js](https://github.com/okta/okta-oidc-js/tree/master/packages/jwt-verifier), which you  can use to validate Okta access tokens (issued by Okta authorization servers).
+
+```bash
+npm install @okta/jwt-verifier@0.0.14 --save
```
Comment (mraible) [inline _source/_posts/2019-02-14-modern-token-authentication-in-node-with-express.md]:
```suggestion
npm install @okta/jwt-verifier@0.0.14
```

File: _source/_posts/2019-02-14-modern-token-authentication-in-node-with-express.md [hunk 6]
```diff
@@ -0,0 +1,356 @@
...
+```
+
+Create an `index.js` file in the folder then copy and paste the following code into the file:
+
+```javascript
+const express = require('express');
+const OktaJwtVerifier = require('@okta/jwt-verifier');
+
+const clientId = {yourClientId};
+const oktaDomain = {yourOktaOrgUrl};
```
Comment (mraible) [inline _source/_posts/2019-02-14-modern-token-authentication-in-node-with-express.md]: This should be https://{yourOktaDomain} so it's populated after the user logs in.

File: _source/_posts/2019-02-14-modern-token-authentication-in-node-with-express.md [hunk 7]
```diff
@@ -0,0 +1,356 @@
...
+  } else {
+    // Forbidden
+    res.sendStatus(403);
+  }
+}
+
+app.listen(port, () => console.log(`My App listening on port ${port}!`))
+```
+
+Replace `{yourClientId}` and `{yourOktaOrgUrl}` with your Okta **Client ID** and **Org Url** that you obtained in the Okta application set up steps.
```
Comment (mraible) [inline _source/_posts/2019-02-14-modern-token-authentication-in-node-with-express.md]: Remove the part of this sentence that tells them to replace yourOktaOrgUrl. At least, I think that's what we decided with @nbarbettini a couple weeks ago.

File: _source/_posts/2019-02-14-modern-token-authentication-in-node-with-express.md [hunk 8]
```diff
@@ -0,0 +1,356 @@
...
+```
+
+You should now see the response `You are viewing private profile info`:
+
+{% img blog/node-token-auth/token-response.png alt:"token response" width:"700" %}{: .center-image }
+
+And... That's it! Good job! You have built an Express app that serves protected API endpoints using token authentication. You used Okta as your OAuth provider which manages your custom scopes (and claims), your tokens, etc. You also used the JWT validation middleware (okta's JWT verifier library) which handles validating tokens for you automatically.
```
Comment (mraible) [inline _source/_posts/2019-02-14-modern-token-authentication-in-node-with-express.md]:
```suggestion
And... That's it! Good job! You have built an Express app that serves protected API endpoints using token authentication. You used Okta as your OAuth provider which manages your custom scopes (and claims), your tokens, etc. You also used the JWT validation middleware (Okta's JWT verifier library) which handles validating tokens for you automatically.
```

File: _source/_posts/2019-02-14-modern-token-authentication-in-node-with-express.md (hunk 1 above)
Comment (rdegges) [reply _source/_posts/2019-02-14-modern-token-authentication-in-node-with-express.md]: Good catch.

File: _source/_posts/2019-02-14-modern-token-authentication-in-node-with-express.md (hunk 2 above)
Comment (rdegges) [reply _source/_posts/2019-02-14-modern-token-authentication-in-node-with-express.md]: Yah, good call. I had it in-line and it looked weird :(

File: _source/_posts/2019-02-14-modern-token-authentication-in-node-with-express.md (hunk 3 above)
Comment (rdegges) [reply _source/_posts/2019-02-14-modern-token-authentication-in-node-with-express.md]: Ty ty

File: _source/_posts/2019-02-14-modern-token-authentication-in-node-with-express.md (hunk 4 above)
Comment (rdegges) [reply _source/_posts/2019-02-14-modern-token-authentication-in-node-with-express.md]: Oo! Good catch again =D

File: _source/_posts/2019-02-14-modern-token-authentication-in-node-with-express.md (hunk 7 above)
Comment (rdegges) [reply _source/_posts/2019-02-14-modern-token-authentication-in-node-with-express.md]: Ok.

**Comment by rdegges** [comment]
Thanks @mraible! I made all the changes as suggested, wanna approve?

File: _source/_posts/2019-02-14-modern-token-authentication-in-node-with-express.md [hunk 9]
```diff
@@ -0,0 +1,356 @@
...
+```
+
+Create an `index.js` file in the folder then copy and paste the following code into the file:
+
+```javascript
+const express = require('express');
+const OktaJwtVerifier = require('@okta/jwt-verifier');
+
+const clientId = "{yourClientId}";
+const oktaDomain = "{yourOktaDomain}";
```
Comment (mraible) [inline _source/_posts/2019-02-14-modern-token-authentication-in-node-with-express.md]: I think this needs to be "https://{yourOktaDomain}" to be properly populated, or you could add it below.

Comment (rdegges) [reply _source/_posts/2019-02-14-modern-token-authentication-in-node-with-express.md]: Ahhh. Good catch again :(

**Comment by rdegges** [comment]
@mraible fixed!

File: _source/_posts/2019-02-14-modern-token-authentication-in-node-with-express.md [hunk 10]
```diff
@@ -0,0 +1,356 @@
...
+
+Once you've done this, you should have a header field that looks something like this:
+`Authorization: Basic MG9haW94OGJtc0JLXhIYjNjMWJITVdxVlhrdTMwaDc6MktxRQ1FaTWVhdXBvbWdCOXZiNkNPOXBtMnFjSw`
+
+You then need to make a POST API call to your **Org URL** value (you obtained this in the Okta application setup step) plus `/v1/token` with the header `grant_type=client_credentials`.
+
+Using HTTPie in the terminal, this will look like:
+
+```bash
+http -f POST https://{yourOrgUrl}/oauth2/default/v1/token \
```
Comment (mraible) [inline _source/_posts/2019-02-14-modern-token-authentication-in-node-with-express.md]: This should be {yourOktaDomain} as well.

**Comment by rdegges** [comment]
@mraible fixed! Ty =D

**Pull request merged**